pykafka/producer.py

```diff
@@ -140,6 +141,14 @@ def __init__(self,
         self._update_lock = self._cluster.handler.Lock()
         self.start()
 
+        def cleanup(obj):
+            if obj._running:
+                obj.stop()
```

Comment:
Unfortunately I don't think that method will work. The `with` block will stop the producer as soon as the function returns.